dashboard/modules/serve/serve_agent.py

```diff
@@ -115,8 +115,15 @@ async def delete_serve_application(self, req: Request) -> Response:
     async def put_all_deployments(self, req: Request) -> Response:
         from ray.serve.schema import ServeApplicationSchema
         from ray.serve._private.api import serve_start
+        from pydantic import ValidationError
 
-        config = ServeApplicationSchema.parse_obj(await req.json())
+        try:
+            config = ServeApplicationSchema.parse_obj(await req.json())
+        except ValidationError as e:
+            return Response(
+                status=400,
+                text=repr(e),
```

Comment:
I used `repr` instead of `str` since `str` sometimes fails for Unicode exceptions (see [this StackOverflow thread](https://stackoverflow.com/a/4308203) for more info).